Sources/TensorFlow/Core/LazyTensorTFFunctionBuilder.swift

```diff
@@ -0,0 +1,315 @@
+            outputGroups.append(output.outputCount)
+            outputs += Array((0..<output.outputCount).map {
+                    TF_Output(oper: graphNode, index: Int32($0)) })
+        }
+    }
+
+    deinit {
+        TF_DeleteGraph(cTFGraph)
+        TF_DeleteStatus(status)
+    }
+
+    private func newNodeName(base: String) -> String {
+        let name = "\(base)_\(nodeCounter)"
+        nodeCounter += 1
+        return name
+    }
+
+    private func updateAttribute(
+        _ desc: CTFOperationDescription,
+        _ name: String,
+        _ attrValue: LazyTensorOperation.Attribute
+    ) {
+        switch attrValue {
+        case LazyTensorOperation.Attribute.tensorDataTypeValue(let value):
+            TF_SetAttrType(desc, name, value._cDataType)
+        case LazyTensorOperation.Attribute.boolValue(let value):
+            TF_SetAttrBool(desc, name, value ? 1 : 0)
+        case LazyTensorOperation.Attribute.intValue(let value):
+            TF_SetAttrInt(desc, name, Int64(value))
+        case LazyTensorOperation.Attribute.floatValue(let value):
+            TF_SetAttrFloat(desc, name, value)
+        case LazyTensorOperation.Attribute.doubleValue(let value):
+            TF_SetAttrFloat(desc, name, Float(value))
+        case LazyTensorOperation.Attribute.stringValue(let value):
+            value.utf8CString.withUnsafeBufferPointer { buffer in
+                // utf8CString is null-terminated; TF_SetAttrString wants
+                // non-null-terminated.
+                TF_SetAttrString(
+                    desc, name, buffer.baseAddress, buffer.count - 1)
+            }
+        case LazyTensorOperation.Attribute.intArray(let values):
+            let values64 = values.map { Int64($0) }
+            values64.withUnsafeBufferPointer { buffer in
+                TF_SetAttrIntList(
+                    desc, name, buffer.baseAddress, Int32(buffer.count))
+            }
+        case LazyTensorOperation.Attribute.constTensor(let value):
+            let cTensor = TFE_TensorHandleResolve(
+                value._cTensorHandle, status)
+            checkOk(status)
+            TF_SetAttrTensor(desc, name, cTensor!, status)
+        case LazyTensorOperation.Attribute.tensorDataTypeArray(let values):
+            values.withUnsafeBufferPointer { buffer in
+                buffer.withMemoryRebound(to: TF_DataType.self) {
+                    reboundBuffer in
+                    TF_SetAttrTypeList(
+                        desc, name,
+                        reboundBuffer.baseAddress,
+                        Int32(reboundBuffer.count))
+                }
+            }
+        case LazyTensorOperation.Attribute.optionalTensorShape(let value):
+            if let shape = value  {
+                let dimensions: [Int64] = shape.dimensions.map(Int64.init)
+                dimensions.withUnsafeBufferPointer { buffer in
+                    TF_SetAttrShape(desc, name,
```

Comment:
one argument per line, or put it all on one line.

Reply:
Done.